lua/plenary/path.lua

```diff
@@ -262,6 +262,24 @@ function Path:shorten()
   return shorten(self.filename)
 end
 
+function Path:shorten_len(len)
+  if len then
+    local final_match
+    local final_path = ""
```

Comment:
Is this faster doing something like:

```
final_path_parts = {}

for stuff do
  table.insert(final_path_parts, thingy)
end

return table.concat(final_path_parts, os.sep) .. final_match
```

Reply:
I didn't end up testing for speed, but the table approach ended up being better for other reasons.